vector/src/main/java/im/vector/app/features/media/ImageContentRenderer.kt

```diff
@@ -152,7 +153,11 @@ class ImageContentRenderer @Inject constructor(private val activeSessionHolder:
                     .load(resolvedUrl)
         }
 
-        req.override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL)
+        val maximumScale = (contextView as? PhotoView)?.maximumScale?.toInt() ?: 1
+        val width = min(contextView.resources.displayMetrics.widthPixels, data.width ?: 0).times(maximumScale)
+        val height = min(contextView.resources.displayMetrics.heightPixels, data.height ?: 0).times(maximumScale)
```

Comment:
if I've understood this correctly and this is the image height/width rather than viewport... then we'll break the aspect ratio if we end up using the screen width/height instead of the data width/height (eg causing a landscape photo to become portrait)

it would be safer to always apply the scaling to both dimensions 

```kotlin
// not taking into account any nulls
val width = data.width
val height = data.height

if (data.width > displayMetrics.widthPixels || data.height > displayMetrics.heightPixels) {
  val ratio = data.width / data.height
  width = displayMetrics.widthPixels
  height = (displayMetrics.heightPixels * ratio)
}

req.override(width.times(maxScale), height.times(maxScale)
```






```